eclipse-theia/theia
# search-in-workspace: focus on next and previous search results

#### What it does

Fixes: #12692 
Adds `Search: Focus on Next Result` and `Search: Focus on Previous Result` commands to be able to quickly cycle through search results using `F4` and `Shift+F4` keybindings respectively. This allows the user to go through search results without having to focus on the `search-in-workspace` view and needing to use `up/down` keybindings.

#### How to test


1.  Open Theia Application
2.  Search for a string that has multiple results in the `search-in-workspace` view
3.  Use the command palette (`Ctrl+Shift+P`) and search for `Search: Focus on Next Result` or `Search: Focus on Previous Result`
4.  The `search-in-workspace` view should appear and the next/previous search result should be focused and selected.

OR
1.  Open Theia Application
2.  Search for a string that has multiple results in the `search-in-workspace` view
3.  Use the keybindings `F4` and `Shift+F4` for the `Search: Focus on Next Result` and `Search: Focus on Previous Result` respectively
4.  The `search-in-workspace` view should appear and the next/previous search result should be focused and selected.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
